src/components/App.js

```diff
@@ -1,17 +1,94 @@
-import React from 'react';
+import React from "react";
+import Search from "./Search.js";
+import Films from "./Films.js";
+import FilmImage from "./FilmImage.js";
+import Favourites from "./Favourites.js";
+
+//`http://www.omdbapi.com/?s=${film}&apikey=d213a5b1`
+//`http://www.omdbapi.com/?i=${singleFilmInfo}&apikey=d213a5b1`
 
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.removeFavouriteByIndex = this.removeFavouriteByIndex.bind(this);
+
+    this.state = {
+      filmSearch: "",
+      results: [],
+      favouriteFilm: []
+    };
+  }
+
+  componentDidMount() {}
+
+  handleClick(filmTitle) {
+    console.log(filmTitle);
+
+    const favouritesCopy = this.state.favouriteFilm.splice(0);
```

Comment:
Splice mutates the original array by removing items from it and returning them. My guess you want to use slice here to create a copy